app/src/main/java/com/zulip/android/networking/util/DefaultCallback.java

```diff
@@ -0,0 +1,36 @@
+package com.zulip.android.networking.util;
+
+import android.support.annotation.CallSuper;
+import android.util.Log;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+/**
+ * Created by patrykpoborca on 8/25/16.
+ */
+
+public class DefaultCallback<T> implements Callback<T> {
+
+    @Override
+    @CallSuper
+    public void onResponse(Call<T> call, Response<T> response) {
+        if(response.isSuccessful()) {
+            onSuccess(call, response);
+        }
+        else {
+            onFailure(call, null);
+        }
+    }
+
+    public void onSuccess(Call<T> call, Response<T> response) {
+
+    }
+
+    @Override
+    public void onFailure(Call<T> call, Throwable t) {
+        //log error
+        Log.e("DefaultCallback", "Nework failure", t);
```

Comment:
Maybe use `Zlog` so that we receive the error in Crashlytics? This might be too noisy tho :)